src/main/java/frc/robot/subsystems/Shooter/commands/Shoot.java

```diff
@@ -0,0 +1,42 @@
+package frc.robot.subsystems.Shooter.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc.robot.subsystems.Shooter.Shooter;
+
+/**
+ * this command activates the shooter subsystem based on power
+ */
+public class Shoot extends CommandBase {
+    private Shooter shooter;
+    private double velocity;
+
+    /**
+     * Command Constructor
+     * @param shooter - the Shooter Subsystem
+     * @param velocity - target velocity (m/s)

```

Comment:
you should get distance not velocity, and create a function that gets the distance (we get the distance from vision).